plugins/chat/app/jobs/regular/chat/notify_watching.rb

```diff
@@ -83,6 +95,18 @@ def send_notifications(membership)
           channel_id: @chat_channel.id,
         }
 
+        if @chat_message.in_thread? && !membership.muted?
+          thread_membership =
+            ::Chat::UserChatThreadMembership.find_by(
+              user_id: user.id,
+              thread_id: @chat_message.thread_id,
+            )
+
+          if thread_membership&.notification_level == "watching"
+            create_watched_thread_notification(thread_membership)
+          end
```

Comment:
How about moving the "notification_level" condition directly inside the SQL query?

```suggestion
          thread_membership =
            ::Chat::UserChatThreadMembership.find_by(
              user_id: user.id,
              thread_id: @chat_message.thread_id,
              notification_level: "watching",
            )
            thread_membership && create_watched_thread_notification(thread_membership)
```